redfish/internal/controller/http/v1/handler/routes.go

```diff
@@ -21,39 +22,46 @@ import (
 var (
 	metadataXML    string
 	metadataLoaded bool
+	metadataMutex  sync.Mutex
 )
 
 // loadMetadata loads metadata.xml from the generated folder with XML validation.
+// This function is thread-safe and ensures metadata is loaded only once.
 func loadMetadata() {
+	metadataMutex.Lock()
+
+	// Double-check locking pattern
 	if metadataLoaded {
+		metadataMutex.Unlock()
```

Comment:
better way in go to do a defer metadataMutex.Unlock() as soon as you do the metadataMutex.Lock(), so that there's no need to unlock for all the cases.